src/cargo/core/workspace.rs

```diff
@@ -0,0 +1,421 @@
+use std::collections::hash_map::{HashMap, Entry};
+use std::collections::BTreeMap;
+use std::path::{Path, PathBuf};
+use std::slice;
+
+use core::{Package, VirtualManifest, EitherManifest, SourceId};
+use core::{PackageIdSpec, Dependency};
+use ops;
+use util::{Config, CargoResult};
+use util::paths;
+
+/// The core abstraction in Cargo for working with a workspace of crates.
+///
+/// A workspace is often created very early on and then threaded through all
+/// other functions. It's typically through this object that the current
+/// package is loaded and/or learned about.
+pub struct Workspace<'cfg> {
+    config: &'cfg Config,
+    current_manifest: PathBuf,
+    packages: Packages<'cfg>,
+    root_manifest: Option<PathBuf>,
+    members: Vec<PathBuf>,
+}
```

Comment:
Can you add a comment explaining the difference between `current_manifest`, `root_manifest`, and `members`?